server/src/test/java/org/elasticsearch/common/xcontent/XContentTypeTests.java

```diff
@@ -121,4 +121,21 @@ public void testVersionParsing() {
         assertThat(XContentType.parseVersion("APPLICATION/JSON"),
             nullValue());
     }
+
+    public void testVersionParsingOnText() {
```

Comment:
as per my comment https://github.com/elastic/elasticsearch/pull/54103/files#r409392721
maybe we should also extend parsing media types that are text? Should we unify this with SQL?
see `RestSqlQueryAction` at the moment sql has to use `format` parameter
cc @astefan

Reply:
Can we postpone that decision for now and work and in future work ensure that text based APIs are good ?